### PR TITLE
Emit new challenge code for each `create_identity`

### LINKED ISSUE #981: [identityManagement] Support generating new challenge codes for non-verified Identities.
### Context
Pallet: identityManagement[Mock] 

Consider the following scenario
1. A user request `identityManagement.linkIdentity('Twitter', 'my-handle')` -> `ChallengeCodeGenerated` event is emitted.
1. User request again `identityManagement.linkIdentity('Twitter', 'my-handle')`.

**Received:**
The transaction fails with `IdentityAlreadyExist`.

**Desired:**
On point 2, generate a new challenge code. i.e., a `ChallengeCodeGenerated` event is emitted.

## How this useful

- Better Developer Experience: It makes testing easier as you can re-use the same `web_type` and `handle` to get the same results while not minding spending tokens.
- Better User Experience: Users won't need to pay fees for `unlinkIdentity` in case they lose their challengeCode, and retrieving through an indexer is not an option for them.
- Better Consumer Experience: as a dApp developer, I would like to leave ChallengeCodes retrieval from an archive node for later.

### PR DESCRIPTION
**Pull request by Kailai-Wang**
resolves #981 

More test cases for IMP mock and adjustment in sgx-runtime pallets.
some leftover renamings too

TODO: #1073

### CODE REVIEW

File: pallets/identity-management-mock/src/lib.rs [hunk 1]
```diff
@@ -310,14 +310,17 @@ pub mod pallet {
 				},
 			};
 
-			ensure!(
-				!IDGraphs::<T>::contains_key(&who, &identity),
-				Error::<T>::IdentityAlreadyExist
-			);
+			if let Some(c) = IDGraphs::<T>::get(&who, &identity) {
```
Comment (ziming-zung) [inline pallets/identity-management-mock/src/lib.rs:313]: I think we should do the same changes for the `identity-managment-pallet`in the tee-worker, include renamings

https://github.com/litentry/litentry-parachain/blob/c485920fbdc980fcf0a8aac1ace9011c6606c8d1/tee-worker/litentry/pallets/identity-management/src/identity_context.rs#L28-L37

https://github.com/litentry/litentry-parachain/blob/c485920fbdc980fcf0a8aac1ace9011c6606c8d1/tee-worker/litentry/pallets/identity-management/src/lib.rs#L184-L204

and so on....

**Review by ziming-zung — CHANGES_REQUESTED** [review]

File: pallets/identity-management-mock/src/lib.rs (hunk 1 above)
Comment (Kailai-Wang) [reply pallets/identity-management-mock/src/lib.rs:313]: I think it's already done in this PR? 😂 
https://github.com/litentry/litentry-parachain/pull/1074/files#diff-0673e79aa46adac84174d588afcb0816746db60d4b560b4cdd01097951b322b0

Comment (ziming-zung) [reply pallets/identity-management-mock/src/lib.rs:313]: My fault, i should be ok.😂

**Review by ziming-zung — APPROVED** [review]
Looks good.

**Pull request merged**